packages/xo-server/src/xo-mixins/vm.js

```diff
@@ -0,0 +1,19 @@
+export default class Vms {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async revertVm({ snapshot, snapshotBefore, userId }) {
+    const { _xo } = this
+    const snapshotXapi = _xo.getXapi(snapshot)
+    if (snapshotBefore) {
+      const { $id: snapshotId } = await snapshotXapi.snapshotVm(
+        snapshot.$snapshot_of
+      )
+      if ((await _xo.getUser(userId)).permission !== 'admin') {
+        await _xo.addAcl(userId, snapshotId, 'admin')
```

Comment:
ACLs are proper to XO not to XAPI.

Reply:
I don't undertand, Why you mentionned that ACLS are proper to XO?
Here we are on `xo-mixins` not `xapi-mixins`.

Reply:
IMHO, `revertVm` should not be done in `xo-mixins` because it's not proper to XO, and `snapshotBefore` functionality is already integrated to `xapi.revertVm` implementation.

As solution, i suggest to let `xapi.revertVm` returns the created snapshot UUID and you can easily add an ACL for it.